migration-engine/core/src/state.rs

```diff
@@ -56,14 +56,16 @@ impl EngineState {
         path: &str,
         f: ConnectorRequest<O>,
     ) -> CoreResult<O> {
+        let config_dir = std::path::Path::new(path).parent();
         let schema = std::fs::read_to_string(path)
             .map_err(|err| ConnectorError::from_source(err, "Falied to read Prisma schema."))?;
-        self.with_connector_for_schema(&schema, f).await
+        self.with_connector_for_schema(&schema, config_dir, f).await
     }
 
     async fn with_connector_for_schema<O: Send + 'static>(
         &self,
         schema: &str,
+        config_dir: Option<&Path>,
```

Comment:
The optionality is problematic, but we'll have to change the JSON-RPC API for some methods to fix that. Could we persist that somewhere?